orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy

```diff
@@ -115,15 +125,17 @@ class MonitorKatoTask implements RetryableTask, CloudProviderAware {
     def katoResultExpected = (stage.context["kato.result.expected"] as Boolean) ?: false
     ExecutionStatus status = katoStatusToTaskStatus(katoTask, katoResultExpected)
 
-    if (status != ExecutionStatus.TERMINAL && status != ExecutionStatus.SUCCEEDED) {
-      status = ExecutionStatus.RUNNING
-    }
-
     def outputs = [
+      'kato.task.terminalRetryCount': 0,
       'kato.task.firstNotFoundRetry': -1L,
-      'kato.task.notFoundRetryCount': 0
+      'kato.task.notFoundRetryCount': 0,
+      'kato.task.lastStatus': status
     ] as Map<String, ?>
 
+    if (status != ExecutionStatus.TERMINAL && status != ExecutionStatus.SUCCEEDED) {
+      status = ExecutionStatus.RUNNING
+    }
+
```

Comment:
looking at `katoStatusToTaskStatus` i don't even understand why we need this `if`... anyway, either way, you can move it back where it came from

Reply:
Yah, good point. 🤷‍♂